airflow/api_connexion/endpoints/import_error_endpoint.py

```diff
@@ -85,38 +83,30 @@ def get_import_errors(
     """Get all import errors."""
     to_replace = {"import_error_id": "id"}
     allowed_sort_attrs = ["import_error_id", "timestamp", "filename"]
-    count_query = select(func.count(ParseImportError.id))
     query = select(ParseImportError)
     query = apply_sorting(query, order_by, to_replace, allowed_sort_attrs)
 
-    can_read_all_dags = get_auth_manager().is_authorized_dag(method="GET")
-
-    if not can_read_all_dags:
```

Comment:
I think we need to keep something like this unfortunately - otherwise the performance of the query plummets if there are a large number of dags in the deployment.

(We didn't originally have this in the first few releases of per dag access control, and had to add it)

Reply:
:'(

Reply:
Yes. This could be mostly optimisation. I think it's a viable case to have a special case for "ALL" Dags. Simillarly in the future when we get "multi-team" support, we will have to introduce "ALL DAGS FOR TEAM" as a single value - we do not want to keep list of dag_ids that the user can have access to.

In fact that brings me to another case. In case of non-FAB we do not have the notion of allowing access to particular DAGs at all. This is all driven by "whatever the auth manager returns". So we could introduce it as explicit API of auth manager:

* get_permitted_dags = "List of dags I have access to" | ALL | NONE | TEAM  (TEAM in the future). That will automatically allow Auth Manager to handle it in the right "optimized" way

Reply:
That is fair. I could do that. That way we would use `get_permitted_dags` to figure whether a user has access to all DAGs and we could use `is_authorized_dag("GET")` to check whether a user is allowed to list DAGs

Reply:
I updated the PR to implement this proposal. Could you take a look please?

Reply:
Works for me.